pandas/io/parsers.py

```diff
@@ -1009,6 +999,10 @@ def _create_index(self, ret):
     def get_chunk(self, size=None):
         if size is None:
             size = self.chunksize
+        if self.nrows is not None:
+            if self._currow >= self.nrows:
```

Comment:
Because a parser's ``.read()`` raises (or at least, can legitimately raise) ``StopIteration`` only when the current row increases above ``nrows``, and this never happens if you keep asking for 0 rows at a time. This is not hypothetical - the "``# With nrows``" step of ``test_read_chunksize`` in ``pandas/tests/io/parser/common.py`` hangs for the ``PythonParser`` (and possibly others) if I change the code as you suggest.